tests/integration/build_test.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -g cloned_kernel_tree_path_host
+declare -g target_random_distro
+
+function oneTimeSetUp()
+{
+  local distro
+  local container
+  local url_kernel_repo_tree='https://github.com/torvalds/linux'
+
+  # Select a random distro for the tests
+  target_random_distro=$(select_random_distro)
+  cloned_kernel_tree_path_host="$(mktemp --directory)/linux"
+  container="kw-${target_random_distro}"
+
+  # Install kernel build dependencies
+  container_exec "$container" 'yes | ./setup.sh --install-kernel-dev-deps'
+  if [[ "$?" -ne 0 ]]; then
+    complain "Failed to install kernel build dependencies for ${distro}"
+    return 1 # EPERM
+  fi
+
+  eval "git clone --depth 1 ${url_kernel_repo_tree} ${cloned_kernel_tree_path_host}"
```

Comment:
We have a build feature named `--from-sha`, and to test it, we might use a large depth, maybe increasing it to 5.

Reply:
Why do you need eval here? Why not use git directly?